algoliasearch-core/src/test/java/com/algolia/search/integration/client/MultiClusterManagementTest.java

```diff
@@ -48,16 +49,32 @@ void mcmTest() throws ExecutionException, InterruptedException {
 
     removeUserId(mcmClient, userID);
 
-    ListUserIdsResponse listUserIds2 = mcmClient.listUserIDsAsync().get();
+    removePastUserIDs(mcmClient);
+  }
+
+  void removePastUserIDs(SearchClient mcmClient) throws ExecutionException, InterruptedException {
+
+    int page = 0;
+    int hitsPerPage = 100;
+    List<UserId> userIDsToRemove = new ArrayList<>();
+
+    while (true) {
+      ListUserIdsResponse listUserIds = mcmClient.listUserIDsAsync(page, 100, null).get();
 
-    String yesterday =
-        DateTimeFormatter.ofPattern("yyyy-MM-dd-HH")
-            .format(ZonedDateTime.now(ZoneOffset.UTC).minusDays(1));
+      String today =
+          DateTimeFormatter.ofPattern("yyyy-MM-dd-HH").format(ZonedDateTime.now(ZoneOffset.UTC));
 
-    List<UserId> userIDsToRemove =
-        listUserIds2.getUserIDs().stream()
-            .filter(r -> r.getUserID().contains("java-" + yesterday))
-            .collect(Collectors.toList());
+      List<UserId> userIDsTemp =
+          listUserIds.getUserIDs().stream()
+              .filter(r -> !r.getUserID().contains("java-" + today))
```

Comment:
You first need to filter out non `java-` prefixed userIDs first, otherwise, you'll remove userIDs of other API clients (see [the Go implementation](https://github.com/algolia/algoliasearch-client-go/blob/master/cts/search/mcm_test.go#L104-L105) for instance where we first take all `go-` prefixed userIDs and then remove all the `go-{today}` ones).

Reply:
All good, thanks for catching it. Went to fast.